app/pages/lab/collaborators.cjsx

```diff
@@ -225,7 +240,7 @@ module.exports = createReactClass
         user_id: parseInt(projectRoleSet.links.owner.id)
         section: @talkSection()
         name: POSSIBLE_ROLES[role]
-      ).save()
+      ).save() unless role in ROLES_NOT_IN_TALK_API 
```

Comment:
does this have to return a role for use in `talkRoleAction` later on?

Reply:
In this bit, if the toggled `role` is `'museum'`, then `talkRoleAction` just becomes `undefined`, so no interaction with Talk occurs whatsoever.

Sorry, CoffeeScript is a wonk to read... this basically translates to...

```
if (index === -1) {
  projectRoleSet.roles.push(role)
  if (!ROLES_NO_IN_TALK_API.contains(role)) {
    talkRoleAction = talkClient.type('roles').create(...).save()
  } else {
    talkRoleAction = undefined
  }
}  ...
```

Reply:
it's the `undefined` that i was wondering about where it'll be used later via `talkRoleAction` https://github.com/zooniverse/Panoptes-Front-End/blob/3d1a78f8012c5401b92d6b5dc357b096be6ea8bb/app/pages/lab/collaborators.cjsx#L250

I'm not really sure how that later code handles undefined...

Reply:
The coffeescript is hard to read but I don't see the later code doing anything with the return values from those promises, so I think it ignores `talkRoleAction` (unless the promise rejects.)